src/main/java/seedu/address/ui/IssueCard.java

```diff
@@ -43,10 +43,14 @@ public IssueCard(Issue issue, int displayedIndex) {
         super(FXML);
         this.issue = issue;
         id.setText(displayedIndex + ". ");
-        description.setText(issue.getDescription().toString() + issue.getId().uiRepresentation());
+        description.setText(issue.getDescription().toString() + " " + issue.getIssueId().uiRepresentation());
+        if (issue.getDeadline().isEmpty()) {
+            deadline.setText("No Deadline Set");
```

Comment:
Adding a method for the display string representation in each attribute class may be a better option, this way you can override the display method in the singleton class and will not have to do the checks here